CDS/Collections/TernarySearchTrie.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CDS.Collections
+{
+    /**
+     * Ternary Search Trie (TST) based on Sedwick and Wayne, Algorithms Fourth Edition's TST.
+     * Note that this implementation MUST be used with non-primitive types. You may not use int, char, string etc.
+     * Use the Wrapper classes (That returns null as default value! Int32 does not return null) if you need to use them.
+     **/
+    public class TernarySearchTrie<Value> : ITrie<Value>
```

Comment:
Use proper C# naming convetions regarding generic types: `TValue` instead of `Value`